drivers/syslog/syslog_channel.c

```diff
@@ -129,6 +129,16 @@ static struct syslog_channel_s g_default_channel =
 };
 #endif
 
+/* This is a simply sanity check to avoid we have more elements than the
+ * `g_syslog_channel` array can hold
+ */
+
+#if (CONFIG_SYSLOG_DEFAULT + CONFIG_RAMLOG_SYSLOG + \
+     CONFIG_SYSLOG_RPMSG + CONFIG_SYSLOG_RTT) > \
+     CONFIG_SYSLOG_MAX_CHANNELS
+#  error "Max syslog channel number exceeds"
```

Comment:
how about change to static_assert?

Reply:
think this will generate warnings if any of `CONFIG_SYSLOG_DEFAULT`, `CONFIG_RAMLOG_SYSLOG`, `CONFIG_SYSLOG_RPMSG` or `CONFIG_SYSLOG_RTT` are not defined.

Reply:
Sure, will have a look at changing to `static_assert` method.  :-)

Reply:
> think this will generate warnings if any of `CONFIG_SYSLOG_DEFAULT`, `CONFIG_RAMLOG_SYSLOG`, `CONFIG_SYSLOG_RPMSG` or `CONFIG_SYSLOG_RTT` are not defined.

True, will have a think about a better solution. Thx!

Reply:
Another question: is it a good idea to let `SYSLOG_MAX_CHANNELS` be determined automatically by filling `CONFIG_SYSLOG_DEFAULT`, `CONFIG_RAMLOG_SYSLOG`, etc. into an `enum` data type?

Reply:
It doesn't work as expect, because the user may add additional channel dynamically.